bundle/config/root.go

```diff
@@ -138,6 +138,11 @@ func (r *Root) updateWithDynamicValue(nv dyn.Value) error {
 	// Assign the normalized configuration tree.
 	r.value = nv
 
+	err = r.Resources.VerifyAllResourcesDefined()
+	if err != nil {
+		return err
+	}
```

Comment:
Can we run this from a validation mutator that we run as part of initialize instead?

Would like to minimize the amount of logic we add to the path executed for every mutator.

Reply:
@pietern at the moment it has to be done inside `updateWithDynamicValue` because otherwise ConfigureConfigFilePath will fail with a panic. In the future, we should move this check to a separate mutator in initialise phase. I will add a comment for this in the code

Reply:
Thanks -- makes sense!